exporter/datadogexporter/internal/logs/translator.go

```diff
@@ -0,0 +1,166 @@
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/otlp/model/attributes"
+	"github.com/DataDog/datadog-agent/pkg/otlp/model/source"
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadog"
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	conventions "go.opentelemetry.io/collector/semconv/v1.6.1"
+)
+
+const (
+	namespace          = "otel"
+	otelTraceID        = namespace + ".trace_id"
+	otelSpandID        = namespace + ".span_id"
+	otelSeverityNumber = namespace + ".serverity_number"
+	otelSeverityText   = namespace + ".serverity_text"
+	otelTimestamp      = namespace + ".timestamp"
+
+	ddStatus    = "status"
+	ddTraceID   = "dd.trace_id"
+	ddSpanID    = "dd.span_id"
+	ddTimestamp = "@timestamp"
+)
+
+const (
+	LogLevelTrace = "trace"
+	LogLevelDebug = "debug"
+	LogLevelInfo  = "info"
+	LogLevelWarn  = "warn"
+	LogLevelError = "error"
+	LogLevelFatal = "fatal"
+)
+
+// Transform is responsible to convert LogRecord to datadog format
+func Transform(lr plog.LogRecord, res pcommon.Resource) datadogV2.HTTPLogItem {
+	hostName, serviceName := extractHostNameService(res)
+
+	l := datadogV2.HTTPLogItem{
+		AdditionalProperties: make(map[string]string),
+	}
+	if hostName != "" {
+		l.Hostname = datadog.PtrString(hostName)
+	}
+	if serviceName != "" {
+		l.Service = datadog.PtrString(serviceName)
+	}
+
+	// we need to set log attributes as AdditionalProperties
+	lr.Attributes().Range(func(k string, v pcommon.Value) bool {
+		l.AdditionalProperties[k] = v.AsString()
+		return true
+	})
+	if !lr.TraceID().IsEmpty() {
+		l.AdditionalProperties[ddTraceID] = convertTraceID(lr.TraceID().HexString())
+		l.AdditionalProperties[otelTraceID] = lr.TraceID().HexString()
+	}
+	if !lr.SpanID().IsEmpty() {
+		l.AdditionalProperties[ddSpanID] = convertTraceID(lr.SpanID().HexString())
+		l.AdditionalProperties[otelSpandID] = lr.SpanID().HexString()
+	}
+	var status string
+
+	if lr.SeverityText() != "" {
+		status = lr.SeverityText()
+		l.AdditionalProperties[otelSeverityText] = lr.SeverityText()
+	} else {
+		status = deriveStatus(int(lr.SeverityNumber()))
+	}
+	l.AdditionalProperties[ddStatus] = status
+
+	if lr.SeverityNumber() != 0 {
+		l.AdditionalProperties[otelSeverityNumber] = fmt.Sprintf("%d", lr.SeverityNumber())
+	}
+
+	// for datadog to use the same timestamp we need to set the additional property of "@timestamp"
+	if lr.Timestamp() != 0 {
+		// we are retaining the nano second precision in this property
+		l.AdditionalProperties[otelTimestamp] = fmt.Sprintf("%d", lr.Timestamp())
+		l.AdditionalProperties[ddTimestamp] = lr.Timestamp().AsTime().Format(time.RFC3339)
+	}
+
+	// TODO:  in case json parser is enabled , we shouldn't set Message
+	// When we enable json parser, all the log attributes would be added as AdditionalProperties
```

Comment:
I don't understand this comment. What is the JSON parser here? Is there a link that we can add here for context?

Reply:
I have debating this alot, i will be removing this for now .